compiler/rustc_typeck/src/check/method/mod.rs

```diff
@@ -445,7 +445,7 @@ impl<'a, 'tcx> FnCtxt<'a, 'tcx> {
                     // them as well. It's ok to use the variant's id as a ctor id since an
                     // error will be reported on any use of such resolution anyway.
                     let ctor_def_id = variant_def.ctor_def_id.unwrap_or(variant_def.def_id);
-                    tcx.check_stability(ctor_def_id, Some(expr_id), span);
+                    tcx.check_stability(ctor_def_id, Some(expr_id), span, Some(method_name.span));
```

Comment:
What happens if you change this and the change below to pass `method_name.span` on the already existing `span` parameter?

Reply:
Will get an error:
```
error[E0658]: use of unstable library feature 'fmt_internals': internal to format_args!
   --> /Users/rustin/ClionProjects/rust/library/core/src/macros/mod.rs:473:49
    |
473 |     ($dst:expr, $($arg:tt)*) => ($dst.write_fmt($crate::format_args!($($arg)*)))
    |                                                 ^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^
    |
    = help: add `#![feature(fmt_internals)]` to the crate attributes to enable

error: aborting due to previous error

For more information about this error, try `rustc --explain E0658`.
error: could not compile `rustc-demangle`

```

Reply:
I also tried #[allow_internal_unstable(format_args)], but it still reports an error.
```rust
#[macro_export]
#[stable(feature = "rust1", since = "1.0.0")]
#[allow_internal_unstable(format_args)]
macro_rules! write {
    ($dst:expr, $($arg:tt)*) => ($dst.write_fmt($crate::format_args!($($arg)*)))
}
```
```rust
error[E0658]: use of unstable library feature 'fmt_internals': internal to format_args!
   --> /Users/rustin/ClionProjects/rust/library/core/src/macros/mod.rs:474:49
    |
474 |     ($dst:expr, $($arg:tt)*) => ($dst.write_fmt($crate::format_args!($($arg)*)))
    |                                                 ^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^
    |
    = help: add `#![feature(fmt_internals)]` to the crate attributes to enable

error: aborting due to previous error

For more information about this error, try `rustc --explain E0658`.
error: could not compile `rustc-demangle`

```

Reply:
It's reported in stage1.

Reply:
https://github.com/rust-lang/rust/blob/d6d028369be9eb02d219506c0dd662a2f863fc18/compiler/rustc_span/src/lib.rs#L512

It looks like this is because `allow_internal_unstable` is not recognized when we use method_span.